harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -271,75 +498,80 @@ def cleanup_iterator() -> None:
-                        self.context.distributed.size > 1
-                        and not self.context._average_training_metrics
-                    ):
-                        warnings.warn(
-                            "Only the chief worker's training metrics are being reported, due "
-                            "to setting average_training_metrics to False.",
-                            UserWarning,
-                        )
-                elif w.kind == workload.Workload.Kind.COMPUTE_VALIDATION_METRICS:
-                    action = "validation"
-                    response = {
-                        "metrics": self._compute_validation_metrics(),
-                        "stop_requested": self.context.get_stop_requested(),
-                    }
-                elif w.kind == workload.Workload.Kind.CHECKPOINT_MODEL:
-                    action = "checkpointing"
-                    uuid = ""
-                    if self.is_chief:
-                        metadata = {
-                            "determined_version": det.__version__,
-                            "steps_completed": self.steps_completed,
-                            "framework": f"torch-{torch.__version__}",
-                            "format": "pickle",
-                        }
-                        with self.context._core.checkpoint.store_path(metadata) as (
-                            path,
-                            storage_id,
-                        ):
-                            self._save(path)
-                            uuid = storage_id
-                        response = {"uuid": storage_id}
-                    else:
-                        response = {}
-                    uuid = self.context.distributed.broadcast(uuid)
-                    for callback in self.callbacks.values():
-                        callback.on_checkpoint_upload_end(uuid=uuid)
+                min_checkpoint_period = self.env.experiment_config.get_min_checkpoint_period()
+                min_validation_period = self.env.experiment_config.get_min_validation_period()
+                try:
+                    for op in self.core_context.searcher.operations():
+                        self.searcher_op = op
+
+                        searcher_unit = self.core_context.searcher.get_configured_units()
+
+                        # We only support batches for now
+                        if searcher_unit == core.Unit.EPOCHS:
+                            train_unit = Batch(self.as_batches(batches=None, records=None, epochs=op.length))
+                        elif searcher_unit == core.Unit.BATCHES:
+                            train_unit = Batch(op.length)
+                        elif searcher_unit == core.Unit.RECORDS:
+                            train_unit = Batch(self.as_batches(batches=None, records=op.length, epochs=None))
+                        else:
+                            raise ValueError(f"Train unit {searcher_unit} not supported.")
```

Comment:
nit: rephrase this as "unrecognized searcher unit" since you already did an exhaustive test of allowable enum types.  Arriving here is a bug, not a valid searcher unit which we happen to not support.

Reply:
done